### ISSUE TITLE
Removing a used macro parameter breaks deploy

### ISSUE THREAD

**Issue by FransdeJong** [umbraco/Umbraco.Deploy.Issues#53]
If you remove a parameter on a macro that is used in grid in a site you can't restore content anymore.

Reproduction:
- Create a macro with 2 parameters locally
- push to live
- create content on live using the macro in a grid and fill the 2 parameters
- Remove one parameter locally and push to live
- Try to restore the content locally and deploy will fail with the following error:
System.Collections.Generic.KeyNotFoundException with message: Parameter not found: [parameteralias], on macro: [macroname].

If you alter the Uda to contain the deleted parameter again and push that to live the restoring works as is should.

There are 2 possible ways to solve this issue:
1. Tell me on what nodes the parameter still exists so I can resave it although this is impossible when it's used often
2. Ignore the parameter when it doesn't exist anymore. This would be my preferred solution.

Let me know if I can help in any way!

**Comment by AndyButland** [comment]
We've implemented the second suggestion here, as agree this should just be ignored and not throw an exception.

**Comment by TomChancer** [comment]
Hi all,

Just ran into this issue on a v13 site recently with
<img width="561" height="272" alt="Image" src="https://github.com/user-attachments/assets/50578ef3-8499-4656-be74-8f2f74bbc3be" />

This particular site is 13.13.1 (Little older) with Deploy.Cloud on 13.4.5 and Deploy Contrib on 13.3.1 (Also a little older)

Did this get patched then regressed then patched again in a later version for 13? As I need to get past this error to perform a migration to V17.

Thanks!

**Comment by ronaldbarendse** [comment]
Hi @TomChancer, thanks for the detailed report.

This isn't a regression of the original fix, but a gap in it. The fix in Deploy 4.2.0 covered macros added as a dedicated grid "macro" cell, where a removed parameter is now ignored. The error in your screenshot comes from a different code path: macros inserted into a Rich Text Editor (either an RTE property or an RTE grid cell). That path still throws when the stored macro markup contains a parameter that no longer exists on the macro. This applies to all Deploy 13.x versions, including the latest 13.4.5, so updating Deploy or Deploy Contrib won't get you past it.

Umbraco 13 is currently in security phase (and EOL 14 December 2026) and macros were removed from the CMS in v14, so this won't be changed in Deploy 13. There are two ways to get unblocked:

### Option 1: re-add the parameter on the source environment

The error is thrown by the environment you're transferring or restoring *from*. Add an `autoPlay` parameter back to the `YoutubeVideo` macro there (same alias, any type) and retry. Once the content has been moved you can remove it again.

### Option 2: clean up the affected content

Re-insert the macro in the Rich Text Editor on each affected node and save. That rewrites the macro markup without the stale `autoPlay` attribute. To find the nodes, something like this against the source database should point you at them:

```sql
SELECT DISTINCT n.id, n.[text]
FROM umbracoPropertyData pd
JOIN umbracoContentVersion cv ON cv.id = pd.versionId
JOIN umbracoNode n ON n.id = cv.nodeId
WHERE pd.textValue LIKE '%YoutubeVideo%autoPlay%'
```

Option 1 is quicker if the macro is used in many places.

One thing to keep in mind for the v17 migration: macros no longer exist in v14 and later, so the macro markup itself will need converting (for example to a block) as part of the upgrade regardless of how you get past this error.

**Comment by TomChancer** [comment]
Hey @ronaldbarendse 

Ah thanks for pointing that out and the thorough explanation, No problem on it not being fixed, but now I know the source, we can re-add it in to unblock content exports.
Goal is converting all these macros on an import-migration into V17 anyway, so as long as we have a moveable export, that s what was important.
